packages/enzyme-adapter-react-16/src/ReactSixteenAdapter.js

```diff
@@ -534,9 +595,16 @@ class ReactSixteenAdapter extends EnzymeAdapter {
         if (type.displayName) {
           return type.displayName;
         }
-        const name = type.render.displayName || functionName(type.render);
+        const name = displayNameOfNode({ type: type.render });
         return name ? `ForwardRef(${name})` : 'ForwardRef';
       }
+      case Lazy || NaN: {
+        if (type.displayName) {
+          return type.displayName;
+        }
+        const name = displayNameOfNode({ type: type._result });
+        return name ? `React.lazy(${name})` : 'lazy';
```

Comment:
how do the dev tools show a lazy component?

Reply:
After look into what dev tools show I found it either shows the fallback component or the loaded wrapped component. It won't show the lazy component. But in shallow rendering because we keep the lazy component node, this is still used.